documentation/docs/implementation_design/packages_plugins.md

```diff
@@ -1,4 +1,18 @@
-# Dependency of packages and plugins
+---
+description: GoShimmer uses the adapter design pattern to easily switch between different implementations and internal interfaces just by using a different plugin, without having to rewrite the code using it.
+image: replaceWithGoShimmerLogo
+keywords:
+- dependency
+- plugins
+- plugin system
+- code
+- internal logic
+- package
+- adapter design pattern
```

Comment:
does it grab partial queries like "adapter"?

Reply:
It does, I'll add it .
```suggestion
- adapter design pattern
- adapter
```